components/director/internal/domain/api/service.go

```diff
@@ -1,8 +1,158 @@
 package api
 
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"github.com/kyma-incubator/compass/components/director/internal/labelfilter"
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/pkg/errors"
+)
+
+//go:generate mockery -name=APIRepository -output=automock -outpkg=automock -case=underscore
+type APIRepository interface {
+	GetByID(id string) (*model.APIDefinition, error)
+	ListByApplicationID(applicationID string) ([]*model.APIDefinition, error)
+	CreateMany(item []*model.APIDefinition) error
+	List(filter []*labelfilter.LabelFilter, pageSize *int, cursor *string) (*model.APIDefinitionPage, error)
+	Create(item *model.APIDefinition) error
+	Update(item *model.APIDefinition) error
+	Delete(item *model.APIDefinition) error
+	DeleteAllByApplicationID(id string) error
+}
+
 type service struct {
+	repo APIRepository
+}
+
+func NewService(repo APIRepository) *service {
+	return &service{repo: repo}
+}
+
+func (s *service) List(ctx context.Context, filter []*labelfilter.LabelFilter, pageSize *int, cursor *string) (*model.APIDefinitionPage, error) {
+	return s.repo.List(filter, pageSize, cursor)
+}
+
+func (s *service) Get(ctx context.Context, id string) (*model.APIDefinition, error) {
+	api, err := s.repo.GetByID(id)
+	if err != nil {
+		return nil, err
+	}
+
+	return api, nil
+}
+
+func (s *service) Create(ctx context.Context, in model.APIDefinitionInput) (string, error) {
+	id := uuid.New().String()
+
+	api := &model.APIDefinition{
```

Comment:
`APIDefinitionInput` should have method `toAPIDefinition(id)`, which should be called here